extension-style/src/main/java/com/mapbox/maps/extension/style/projection/generated/Projection.kt

```diff
@@ -0,0 +1,166 @@
+// This file is generated.
+
+package com.mapbox.maps.extension.style.projection.generated
+
+import android.util.Log
+import androidx.annotation.UiThread
+import com.mapbox.bindgen.Value
+import com.mapbox.maps.MapboxStyleException
+import com.mapbox.maps.extension.style.StyleContract
+import com.mapbox.maps.extension.style.StyleInterface
+import com.mapbox.maps.extension.style.expressions.generated.Expression
+import com.mapbox.maps.extension.style.layers.properties.PropertyValue
+import com.mapbox.maps.extension.style.layers.properties.generated.Name
+import com.mapbox.maps.extension.style.types.ProjectionDsl
+import com.mapbox.maps.extension.style.utils.unwrap
+
+/**
+ * A style's projection property sets which projection a map is rendered in.
+ *
+ * @see [The online documentation](https://docs.mapbox.com/mapbox-gl-js/style-spec/projection/)
+ */
+@UiThread
+class Projection : ProjectionDslReceiver, StyleContract.StyleProjectionExtension {
+  private var delegate: StyleInterface? = null
+  private val properties = HashMap<String, PropertyValue<*>>()
+
+  /**
+   * The name of the projection to be used for rendering the map.
+   */
+  val name: Name?
+    /**
+     * Get the Name property.
+     *
+     * @return name as Name
+     */
+    get() {
+      return getPropertyValue("name")
+    }
+
+  /**
+   * Set the Name property.
+   *
+   * The name of the projection to be used for rendering the map.
+   *
+   * @param name as Name
+   */
+  override fun name(name: Name) = apply {
+    setProperty(PropertyValue("name", name))
+  }
+
+  /**
+   * This is an Expression representation of "name".
+   *
+   * The name of the projection to be used for rendering the map.
+   */
+  val nameAsExpression: Expression?
+    /**
+     * Get the Name property as an Expression
+     *
+     * @return Name
+     */
+    get() {
+      getPropertyValue<Expression>("name")?.let {
+        return it
+      }
+      name?.let {
+        return Expression.literal(it.value)
+      }
+      return null
+    }
```

Comment:
In what situation would you need to use this?

Reply:
I'm looking at the style spec and [docs](https://docs.mapbox.com/mapbox-gl-js/style-spec/projection/) and it's not clear to me that expressions are even supported for Projection.name

Reply:
let's double check with gl-native team.

Reply:
verified that our basic expressions are supported, e.g setting.:
```
concat {
  literal("glo")
  literal("b")
  literal("e")
}
```
works fine

Reply:
@macdrevx in general we do support setting property as expression everywhere in generated style code, if it does not apply for iOS - I guess it's fine not to expose it